### PR TITLE
Use application.yml for rest-api

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Remove `.env` config file and switch to `application.yml`
- `application.yml` loaded automatically from 3 sources with the latter sources extending (not overwriting) the previous:
  - `rest-api/config/application.yml`
  - `rest-api/application.yml`
  - `${CONFIG_PATH}`
- The `rest-api/config/application.yml` contains the application defaults. It is expected users don't modify this file but create a file at one of the other two paths.
- Hardcoded defaults in config.js like TTLs, limits and log levels were made configurable
- Fixed a hardcoded shard num
- Rename RESPONSE_ROWS to maxLimit

**Which issue(s) this PR fixes**:
Partially addresses #55

**Special notes for your reviewer**:
- Will follow up right after with a Docs PR
- Please turn on ignore whitespace when viewing diff

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: rest-api/__acceptancetests__/accounts.test.js [hunk 1]
```diff
@@ -61,9 +61,9 @@ var acceptanceTestsAccounts = (function () {
         expect(response.status).toEqual(200);
         let accounts = JSON.parse(response.text).accounts;
 
-        expect(accounts.length).toBe(config.limits.RESPONSE_ROWS);
+        expect(accounts.length).toBe(config.api.limits.responseRows);
```
Comment (Nana-EC) [inline rest-api/__acceptancetests__/accounts.test.js]: nit: this gets called so many times I wonder if you can just retrieve it once and use it throughout

File: rest-api/__acceptancetests__/balances.test.js [hunk 2]
```diff
@@ -61,8 +61,8 @@ var acceptanceTestsBalances = (function () {
         expect(response.status).toEqual(200);
         let balances = JSON.parse(response.text).balances;
 
-        expect(balances.length).toBe(config.limits.RESPONSE_ROWS);
-        if (balances.length !== config.limits.RESPONSE_ROWS) {
+        expect(balances.length).toBe(config.api.limits.responseRows);
+        if (balances.length !== config.api.limits.responseRows) {
```
Comment (Nana-EC) [inline rest-api/__acceptancetests__/balances.test.js]: same comment here, pull it out once and use it in the tests.

File: rest-api/__tests__/integration.test.js [hunk 3]
```diff
@@ -114,7 +112,7 @@ const instantiateDatabase = async function() {
 const flywayMigrate = function() {
     console.log('Using flyway CLI to construct schema');
     let exePath = path.join('.', 'node_modules', 'node-flywaydb', 'bin', 'flyway');
-    let configPath = '.node-flywaydb.integration.conf';
+    let configPath = 'config/.node-flywaydb.integration.conf';
```
Comment (steven-sheehy) [inline rest-api/__tests__/integration.test.js]: TODO: Use path.join

File: rest-api/__acceptancetests__/balances.test.js (hunk 2 above)
Comment (steven-sheehy) [reply rest-api/__acceptancetests__/balances.test.js]: Done

File: rest-api/__acceptancetests__/accounts.test.js (hunk 1 above)
Comment (steven-sheehy) [reply rest-api/__acceptancetests__/accounts.test.js]: Done

File: rest-api/config/application.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      fileUpdateRefreshTimes:
+        balances: 900
+        events: 300
+        records: 5
+      includeHostInLink: false
+      maxLimit: 1000
+      log:
+        level: debug
+      port: 5551
+      ttl:
```
Comment (apeksharma) [inline rest-api/config/application.yml]: nit: Let's add 'cache' in the name somewhere.

**Pull request merged**